pkg/telemetry/prom_cass_servicemonitor.go

```diff
@@ -27,7 +25,6 @@ spec:
     path: /metrics
     scheme: http
     scrapeTimeout: 15s
-    targetPort: %d
```

Comment:
OK I think I understand this now. Looking at how `port` and `targetPort` [are handled](https://github.com/prometheus-operator/prometheus-operator/blob/v0.28.0/pkg/prometheus/promcfg.go#L378-L399),  we don't need `targetPort`. `port` most be a named port exposed by the all-pods service, so here `prometheus` is indeed a named port, and corresponds to 9103.

For the new, modern stack, we just need to switch to the new named port: `metrics` which corresponds to 9000.